### ISSUE TITLE
yet another end environment script

### ISSUE THREAD

**Issue by Grumblin** [Adriftus-Studios/network-script-data#82]
on player breaks snow_layer in the end

- [ ] determine how many layers are broken
- [ ] give an amount of 'coins' based on the amount of layers broken, coins will have CMD
- [ ] chance that each layer doesn't give any coins 
- [ ] coins are sell-able in the market shop

**Comment by BehrRiley** [comment]
This is going to require some of the following:
A) A completely different server with a different resource pack, or:
B) A completely different resource pack loaded when loading this world
C) Custom blocks with their own specified model_data
D) Removing all `snow` from servers that utilize the resource pack to favor letting this server be the only place with snow

The last option is likely not what you're implying but there's not an easy way to override that for just one world the way your request implies.

**Comment by ChrispyMC** [comment]
Everything in this feature request is easy to implement, apart from the custom model data, which has to deal with what Behr said above. Using the [`<MaterialTag.level>`](https://one.denizenscript.com/denizen/tags/MaterialTag.level) tag allows us to see what the current level is for levelled materials. This includes liquids, cake, beehives, and, most important of all (in this case), snow. In the world event [`on player breaks block/<material>`](https://one.denizenscript.com/denizen/evts/player%20breaks%20block), we can use `<context.material.level>` to return the level of the snow layers that was broken. Using [`<util.random.int[<#>].to[<#>]>`](https://one.denizenscript.com/denizen/tags/util.random.int) we can fulfill the third outlined requirement. If the random integer is between a certain range, we then use some math to calculate how many coins to give the player. Adding coins as a sellable item to the market (or a shop) is a trivial task, as evident by [Pull Request 58](https://github.com/BehrRiley/network-script-data/pull/58) and [Pull Request 63](https://github.com/BehrRiley/network-script-data/pull/63). Hopefully, we figure out how to implement the resource pack.

**Comment by ChrispyMC** [comment]
Behr, I don't think Devin intends for the coins to be the actual `snow` material. In the world event, we can just `- determine cancelled` for the event `player breaks snow`. We don't need to give snow custom model data. We give that to another item. Custom model data does not apply to *all items* with the same material. It only applies to ***this*** specific Denizen item with *this* specific CMD number.

**TL;DR: We're not applying, and don't have to apply, custom model data to the snow material. Instead we apply it to another material (an item, not a block) through a Denizen item.**

**Comment by Ph4i1ur3** [comment]
There's no way to safely implement this in the described way non-destructively, at the current moment/
